controllers/kustomization_decryptor.go

```diff
@@ -147,24 +148,31 @@ func (kd *KustomizeDecryptor) ImportKeys(ctx context.Context) error {
 		defer os.RemoveAll(tmpDir)
 
 		var ageIdentities []string
-		for name, file := range secret.Data {
+		var vaultToken string
+		for name, value := range secret.Data {
 			switch filepath.Ext(name) {
 			case ".asc":
 				keyPath, err := securejoin.SecureJoin(tmpDir, name)
 				if err != nil {
 					return err
 				}
-				if err := os.WriteFile(keyPath, file, os.ModePerm); err != nil {
+				if err := os.WriteFile(keyPath, value, os.ModePerm); err != nil {
 					return fmt.Errorf("unable to write key to storage: %w", err)
 				}
 				if err := kd.gpgImport(keyPath); err != nil {
 					return err
 				}
 			case ".agekey":
-				ageIdentities = append(ageIdentities, string(file))
+				ageIdentities = append(ageIdentities, string(value))
+			case ".vault-token":
```

Comment:
As `secret.Data` is a map, and the [iteration order of maps is undefined in Go](https://go.dev/blog/maps#iteration-order), this has a chance on picking a random key every time if multiple are defined.

I think this should be changed to either support multiple tokens (and collect them like the age identities above), use a fixed name (e.g. `token.vault`), or at least be stable sorted.

Reply:
We are expecting one key with the `vault-token` extension to be in the secret, but you are right we never know what might happen. I'll go for the fixed key name as I don't think we should accept several tokens.